### PR TITLE
New Animated Network Demo/Example

### PR DESCRIPTION
**Pull request by filipinascimento**
Adds a new animated network demo based on `helios` and billboard rendering. This also illustrates hardware picking for hovering.

Also includes a modification to the `label` function inside `actor.py` to allow access to the `vtkVectorText` so that it is possible to change the text content of the label.

### CODE REVIEW

File: docs/examples/viz_network_animated_billboard.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+
+###############################################################################
+# We add observers to pick the nodes and enable hardware selector
+
+
+nodes_actor.AddObserver('LeftButtonPressEvent', left_click_callback, 1)
+
+hsel = vtk.vtkHardwareSelector()
+hsel.SetFieldAssociation(vtk.vtkDataObject.FIELD_ASSOCIATION_CELLS)
+hsel.SetRenderer(scene)
```
Comment (skoudoro) [inline docs/examples/viz_network_animated_billboard.py:306]: this part of the code needs to be moved in the `picking` module

File: docs/examples/viz_network_animated_billboard.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+
+        nodes_actor.GetMapper().GetInput().GetPoints().GetData().Modified()
+        nodes_actor.GetMapper().GetInput().ComputeBounds()
+        selectedActor.GetMapper().GetInput().ComputeBounds()
+        selectedActor.GetMapper().GetInput().GetPoints().GetData().Modified()
+        showm.scene.ResetCameraClippingRange()
+        showm.render()
+
+        if counter >= max_iterations:
+            showm.exit()
```
Comment (skoudoro) [inline docs/examples/viz_network_animated_billboard.py:251]: why you do not stop the layout here?

File: fury/actor.py [hunk 3]
```diff
@@ -2051,6 +2051,7 @@ def label(text='Origin', pos=(0, 0, 0), scale=(0.2, 0.2, 0.2),
 
     texta.GetProperty().SetColor(color)
     texta.SetPosition(pos)
+    texta.text = atext
```
Comment (skoudoro) [inline fury/actor.py:2054]: a small and nice change that required a test. Could you add it?

**Review by skoudoro — COMMENTED** [review]
Hi @filipinascimento,

Overall, it looks good. It would be good if you could simplify the code to make it more beginner-friendly.

Also, I got sometimes some error like:
```python
Traceback (most recent call last):
  File "viz_network_animated_billboard.py", line 292, in left_click_callback
    selectedActor.SetPosition(positions[selectedNode])
IndexError: index 1326007 is out of bounds for axis 0 with size 10000
Traceback (most recent call last):
  File "viz_network_animated_billboard.py", line 241, in _timer
    selectedActor.SetPosition(positions[selectedNode])
IndexError: index 1326007 is out of bounds for axis 0 with size 10000
Traceback (most recent call last):
  File "viz_network_animated_billboard.py", line 241, in _timer
    selectedActor.SetPosition(positions[selectedNode])
IndexError: index 1326007 is out of bounds for axis 0 with size 10000
```

I did not find the pattern to reproduce this error consistently.

File: docs/examples/viz_network_animated_billboard.py [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+float radius = 1.;
+if(len > radius){
+    discard;
+}
+
+vec3 normalizedPoint = normalize(vec3(point.xy, sqrt(1. - len)));
+vec3 direction = normalize(vec3(1., 1., 1.));
+float ddf = max(0, dot(direction, normalizedPoint));
+float ssf = pow(ddf, 24);
+fragOutput0 = vec4(max(color*0.5+ddf * color, ssf * vec3(1)), 1);
```
[the remark this answers is not in the document]
Comment (filipinascimento) [reply docs/examples/viz_network_animated_billboard.py:167]: Where would be a good place to have the shader files? In a `shaders` folder inside examples? or maybe in another place?

File: docs/examples/viz_network_animated_billboard.py (hunk 2 above)
Comment (filipinascimento) [reply docs/examples/viz_network_animated_billboard.py:251]: The user may close the window before the counter reaches the maximum number of iterations, that is why the stopLayout is not in the timer function. Any suggestions?

File: docs/examples/viz_network_animated_billboard.py [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+
+    Returns
+    -------
+    vertices : vtkarray
+
+    """
+    return actor.GetMapper().GetInput().GetPoints().GetData()
+
+
+def vtk_array_from_actor(actor, array_name):
```
[the remark this answers is not in the document]
Comment (filipinascimento) [reply docs/examples/viz_network_animated_billboard.py]: done

File: docs/examples/viz_network_animated_billboard.py [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+from fury.utils import (get_actor_from_polydata, numpy_to_vtk_colors,
+                        set_polydata_triangles, set_polydata_vertices,
+                        set_polydata_colors, colors_from_actor,
+                        vertices_from_actor, update_actor)
+
+
+###############################################################################
+# Now, let's define some auxiliary functions
+
+def vtk_vertices_from_actor(actor):
```
[the remark this answers is not in the document]
Comment (filipinascimento) [reply docs/examples/viz_network_animated_billboard.py]: done

**Pull request closed without merging**